### PR TITLE
Fix issue with strings in method calls

### LINKED ISSUE #32: Comments break highlighting of variable declaration in classes
As you can see `public byte AltFormCount` is highlighted correctly, wheres `public byte FormPointer` is not. This appears to be related to the comment, since it will be highlighted correctly when the comment is removed.

It is also noteworthy that this problem only occurs inside class definitions, but not inside methods.

screen shot 2015-08-13 at 18 16 43

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request fixes an issue reported in #28:
```cs
class F {
  a = C("1.1 10\n");
  b = 5;
}
```
See the [fix in Ligthshow](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fpchaigno%2Flanguage-csharp%2F766019a1991dedd3b2c0a1d9912d745eb86d24c8%2Fgrammars%2Fc%2523.cson&grammar_text=&code_source=from-text&code_url=&code=class+F+%7B%0D%0A++a+%3D+C%28%221.1+10%5Cn%22%29%3B%0D%0A%7D).

EDIT: Also fixes #32.

### CODE REVIEW

File: spec/grammar-spec.coffee [hunk 1]
```diff
@@ -1,14 +1,36 @@
...
+      runs ->
+        grammar = atom.grammars.grammarForScopeName('source.cs')
+
     it "parses the grammar", ->
-      grammar = atom.grammars.grammarForScopeName("source.cs")
       expect(grammar).toBeDefined()
       expect(grammar.scopeName).toBe "source.cs"
 
+    it "tokenizes strings in method calls", ->
+      tokens = grammar.tokenizeLines("""
```
Comment (50Wliu) [inline spec/grammar-spec.coffee]: The parentheses here aren't needed.

**Comment by 50Wliu** [comment]
So if I'm understanding this correctly, was the issue that the entire `a = C` part was getting marked as `meta.method.source.cs`?

**Comment by pchaigno** [comment]
> So if I'm understanding this correctly, was the issue that the entire a = C part was getting marked as meta.method.source.cs?

Yep, exactly!

File: spec/grammar-spec.coffee [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
+      expect(tokens[1][3]).toEqual value: '"', scopes: ['source.cs', 'meta.class.source.cs', 'meta.class.body.source.cs', 'meta.method-call.source.cs', 'string.quoted.double.source.cs', 'punctuation.definition.string.begin.source.cs']
+      expect(tokens[1][4]).toEqual value: '1.1 10', scopes: ['source.cs', 'meta.class.source.cs', 'meta.class.body.source.cs', 'meta.method-call.source.cs', 'string.quoted.double.source.cs']
+      expect(tokens[1][5]).toEqual value: '\\n', scopes: ['source.cs', 'meta.class.source.cs', 'meta.class.body.source.cs', 'meta.method-call.source.cs', 'string.quoted.double.source.cs', 'constant.character.escape.source.cs']
+      expect(tokens[1][6]).toEqual value: '"', scopes: ['source.cs', 'meta.class.source.cs', 'meta.class.body.source.cs', 'meta.method-call.source.cs', 'string.quoted.double.source.cs', 'punctuation.definition.string.end.source.cs']
+      expect(tokens[1][7]).toEqual value: ')', scopes: ['source.cs', 'meta.class.source.cs', 'meta.class.body.source.cs', 'meta.method-call.source.cs', 'punctuation.definition.method-parameters.end.source.cs']
+
+
```
Comment (50Wliu) [inline spec/grammar-spec.coffee]: Extra newline here

**Comment by 50Wliu** [comment]
I'm a bit uncomfortable with the notion of checking that a `=` isn't there.  Would it be possible to instead just make sure there's only spaces between the letter and the parentheses, eg `(?!new)[\\w<]\\s*\\(`?  I'm not seeing the point of requiring a space and then another non-equals sign (`\\s+[^=]+`).

**Comment by pchaigno** [comment]
It is the regex for the method definitions. We need the `=` to make sure it's not an assignment. We need the space between the two groups to make sure it's not just a method call. We need to match the whole function definition and not just the parenthesis because afterwards, sub-regex captures the return type, modifiers and the name of the method.

To fix #32, we also need to make sure there isn't any `//` or `/*`.

In short, we need to match:
```cs
void func()
dictionary<int, string> func()
void func(test = default_value)
```
but not:
```cs
a = func(1)
call_to_method()
byte end; //(
byte end; /*(*/
call_to_method  ()
call_to_method ()
```

I think we could "simplify" the whole regex by grouping the things we don't want at the beginning with a negative lookahead. My last commit implements that.

**Comment by 50Wliu** [comment]
:articulated_lorry:
